x/govshuttle/genesis.go

```diff
@@ -5,23 +5,25 @@ import (
 	"github.com/Canto-Network/Canto/v7/x/govshuttle/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	"github.com/ethereum/go-ethereum/common"
 )
 
 // InitGenesis initializes the capability module's state from a provided genesis
 // state.
 func InitGenesis(ctx sdk.Context, k keeper.Keeper, accountKeeper authkeeper.AccountKeeper, genState types.GenesisState) {
 	// this line is used by starport scaffolding # genesis/module/init
 	k.SetParams(ctx, genState.Params)
-
+	k.SetPort(ctx, common.HexToAddress(genState.PortAddress))
```

Comment:
Even when the `PortAddress` is empty, it's converted to a string, which sets it as `0x0000000000000000000000000000000000000000`. This can lead to premature 'found' processing in [this logic](https://github.com/b-harvest/Canto/blob/6a40b399024f825370c9e1692330cbda7af6f01a/x/govshuttle/keeper/proposals.go#L29-L32). When dealing with an empty address, it seems we shouldn't perform the 'set' operation on InitGenesis. Similarly, during export, if it's empty, appropriate handling should be implemented. When converting genState's portAddress to a string like this, the intuitive handling of emptiness isn't straightforward, and such mistakes can occur. Thus, It could be considered to apply the value as []byte directly

Reply:
have fixed the issue by storing port address as []byte https://github.com/b-harvest/Canto/pull/50/commits/2a362644fb7e7958ab2388bfd0d51d444d434760